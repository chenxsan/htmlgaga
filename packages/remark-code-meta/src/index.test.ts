import remarkCodeMeta from './index'
import remark from 'remark'
import remarkHTML from 'remark-html'
interface Callback {
  (err: Error, { contents: string }): void
}

// ```js filename="index.js" filename="C:\\hello world\\"

describe('remarkCodeMeta', () => {
  describe('filename without whitespace', () => {
    const transform = (callback: Callback, options = {}) => {
      remark().use(remarkCodeMeta, options).use(remarkHTML).process(
        `
\`\`\`js filename=index.js
const answer = 42;
\`\`\``,
        callback
      )
    }
    test('should transform code with meta', () => {
      transform(function (err, { contents }) {
        expect(err).toBeNull()
        expect(contents).toMatchSnapshot()
      })
    })
    test('should customize className', () => {
      transform(
        function (err, { contents }) {
          expect(err).toBeNull()
          expect(contents).toContain('voidjs__filename')
          expect(contents).toMatchSnapshot()
        },
        {
          className: 'voidjs',
        }
      )
    })
    test('should customize useDetails', () => {
      transform(
        function (err, { contents }) {
          expect(err).toBeNull()
          expect(contents).not.toContain('details')
          expect(contents).not.toContain('summary')
          expect(contents).toMatchSnapshot()
        },
        {
          useDetails: false,
        }
      )
    })
  })
  describe('filename with whitespace', () => {
    const transform = (callback: Callback, options = {}) => {
      remark().use(remarkCodeMeta, options).use(remarkHTML).process(
        `
\`\`\`js filename="C:\\code meta\\index.js"
const answer = 42;
\`\`\``,
        callback
      )
    }
    test('should transform code with meta', () => {
      transform(function (err, { contents }) {
        expect(err).toBeNull()
        expect(contents).toMatchSnapshot()
      })
    })
  })

  describe('filename and open', () => {
    const transform = (callback: Callback, options = {}) => {
      remark().use(remarkCodeMeta, options).use(remarkHTML).process(
        `
\`\`\`js filename=index.js open=false
const answer = 42;
\`\`\``,
        callback
      )
    }
    test('should transform code with open set to false', () => {
      transform(function (err, { contents }) {
        expect(err).toBeNull()
        expect(contents).not.toContain('open')
        expect(contents).toMatchSnapshot()
      })
    })
  })
})
