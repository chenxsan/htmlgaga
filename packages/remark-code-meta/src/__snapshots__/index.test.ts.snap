// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remarkCodeMeta filename and open should transform code with open set to false 1`] = `
"<details class=\\"remark-code-meta\\"><summary class=\\"remark-code-meta__filename\\">index.js</summary><pre><code class=\\"language-js\\" data-language=\\"js\\">const answer = 42;
</code></pre></details>
"
`;

exports[`remarkCodeMeta filename with whitespace should transform code with meta 1`] = `
"<details class=\\"remark-code-meta\\" open><summary class=\\"remark-code-meta__filename\\">C:\\\\code meta\\\\index.js</summary><pre><code class=\\"language-js\\" data-language=\\"js\\">const answer = 42;
</code></pre></details>
"
`;

exports[`remarkCodeMeta filename without whitespace should customize className 1`] = `
"<details class=\\"voidjs\\" open><summary class=\\"voidjs__filename\\">index.js</summary><pre><code class=\\"language-js\\" data-language=\\"js\\">const answer = 42;
</code></pre></details>
"
`;

exports[`remarkCodeMeta filename without whitespace should customize useDetails 1`] = `
"<div class=\\"remark-code-meta\\"><div class=\\"remark-code-meta__filename\\">index.js</div><pre><code class=\\"language-js\\" data-language=\\"js\\">const answer = 42;
</code></pre></div>
"
`;

exports[`remarkCodeMeta filename without whitespace should transform code with meta 1`] = `
"<details class=\\"remark-code-meta\\" open><summary class=\\"remark-code-meta__filename\\">index.js</summary><pre><code class=\\"language-js\\" data-language=\\"js\\">const answer = 42;
</code></pre></details>
"
`;
